### ISSUE TITLE
[WFRP4e Core] Spells

### ISSUE THREAD

**Issue by Txus5012** [moo-man/WFRP4e-FoundryVTT#791]
I've tested the spells like I did with the prayers. Mainly focusing in applying each effect of each spell.

Lores:
-Invoke Lore of Beasts button doesn't appear in the cast roll.
-Lore of Fire is applied to the caster, not the target.
-Ingredients shouldn't protect from rolling miscasts when using the Lore of Witchcraft. (Well... yeah, I guess this is a tough one)

Spells:
-In Cerulean Shield spell, the Cerulean Damage effect sends an error when attacked.
REPORT
EFFECT:	Cerulean Damage
ACTOR:	Wizard - oJLx0Z269hN4QF0s
ERROR:	TypeError: Cannot read property 'name' of null

-Comet of Cassandora applies the Prone condition, but not the Ablaze condition.

-Destroy Lesser Demon: Characteristic Bonus effect isn't applied. In the WFRP tab the Effect Type must be removed.
Also, a "Foundry note" about how to fill the Effects tab would be useful.

-Distracting does nothing.
Uncaught (in promise) TypeError: Cannot read property 'modifier' of undefined
    at Object.eval (eval at applyOneTimeEffect (utility-wfrp4e.js:795), <anonymous>:3:6)
    at Function.applyOneTimeEffect (utility-wfrp4e.js:796)
    at utility-wfrp4e.js:762
    at Array.forEach (<anonymous>)
    at Function.applyEffectToTarget (utility-wfrp4e.js:760)
    at Function._onApplyEffectClick (chat-wfrp4e.js:602)
    at HTMLElement.dispatch (jquery.min.js:2)
    at HTMLElement.v.handle (jquery.min.js:2)

-Earthblood: Wounds not healing.
Uncaught (in promise) TypeError: Cannot read property 'split' of undefined
    at fromUuid (foundry.js:1528)
    at Object.eval (eval at <anonymous> (actor-wfrp4e.js:3060), <anonymous>:3:1)
    at actor-wfrp4e.js:3061
    at Array.forEach (<anonymous>)
    at ActorWfrp4e.runEffects (actor-wfrp4e.js:3058)
    at checkEndRoundConditions (combat.js:225)
    at combat.js:19
    at Array.forEach (<anonymous>)
    at Function.combatChecks (combat.js:19)
    at preUpdateCombat (combat.js:30)

-Fearsome effect doesn't seem to do anything.
In the colorless version the effect is doubled.

-Feather of Lead: Both effects (Unburdened and Overburdened) show an error.
actor-wfrp4e.js:3066 REPORT
EFFECT:	Feather of Lead - Overburdened
ACTOR:	Dummy - uiLnowVpVLocT0Kd
ERROR:	TypeError: Cannot read property 'state' of undefined
REPORT
EFFECT:	Feather of Lead - Unburdened
ACTOR:	Dummy - uiLnowVpVLocT0Kd
ERROR:	TypeError: Cannot set property 'state' of undefined

-Healing Light: The description says that the Endurance Test is Very Hard (-20), but it should say Very Hard (-30) or Hard (-20) (Actually, this is a bug in the Core Rulebook XD ).

-Steal Life: Maybe it could remove fatigued conditions from attacker? I don't know if that's possible.

-Terryfying: The effect is only added in the colorless version. Invoke Terrifying could be added to the cast roll.

-The First Portent of Amul deletes Fortune points.
Chat message:  [Character] fortune points increased from 0 to NaN

-The Second Portent of Amul can't be rolled.
Uncaught (in promise) TypeError: Cannot read property 'toLowerCase' of undefined
    at ItemWfrp4e.computeSpellPrayerFormula (item-wfrp4e.js:1070)
    at ActorWfrp4e.castTest (actor-wfrp4e.js:1502)

-Transmutation of Chamon only adds the Stunned condition after applying damage, but not Blinded or Deafened.

There are some spells that add conditions that show the EmbeddedCollection error message when removing them. I know you're not fixing them yet, but I'll write them here for completion.
-Great Fires of U’Zhul.
-Steal Life.
-T'Essla's Arc.
-Terryfying.
-The Amber Spear.

**Comment by moo-man** [comment]
Thanks for all the testing, been going through these, got most of them but...

> -Lore of Fire is applied to the caster, not the target.

Can't reproduce

> -Ingredients shouldn't protect from rolling miscasts when using the Lore of Witchcraft. (Well... yeah, I guess this is a tough one)

What do you mean? It specifically says "unless cast with an ingredient"

> -Comet of Cassandora applies the Prone condition, but not the Ablaze condition.

>Transmutation of Chamon only adds the Stunned condition after applying damage, but not Blinded or Deafened.

This is likely due to the database problem with Foundry 0.8

**Comment by Txus5012** [comment]
You're welcome.

-Lore of Fire: I'll check that again.
-Witchcraft: And right after that it says ",where the ingredient provides no further protection should you roll a Miscast."
-Comet and Transmutation: Does it mean that it is a similiar problem than the EmbeddedCollection issue?

**Comment by moo-man** [comment]
Yeah the keyword there is 'further', meaning your ingredient was used to prevent the witchcraft miscast, so if you miscast naturally you'll have to roll.

**Comment by moo-man** [comment]
> Does it mean that it is a similiar problem than the EmbeddedCollection issue

Yes anything that adds embedded documents immediately after each other (Comet adds Prone and Ablaze), generally get "added" and can't be removed, or just don't end up being added at all.

You can read more about it here https://gitlab.com/foundrynet/foundryvtt/-/issues/5309

**Comment by Txus5012** [comment]
> Yeah the keyword there is 'further', meaning your ingredient was used to prevent the witchcraft miscast, so if you miscast naturally you'll have to roll.

Exactly. The issue is that the ingredient also prevents the rolled miscast.
![Witchcraft](https://user-images.githubusercontent.com/87753744/130269052-636a1fd8-5696-4917-b4eb-151504d13ca0.jpg)
Note: This wizard doesn't have any talents to prevent miscasts. In all these examples the spell was fully chanelled.
